scraper/src/Scraper.hs

```diff
@@ -22,7 +23,45 @@ runScrape = do
 scrape :: IO ()
 scrape = do 
   website <- httpLBS "http://leg.colorado.gov/bill-search?field_sessions=10171&sort_bef_combine=field_bill_number%20ASC" 
-  L8.writeFile "test.html" $ getResponseBody website
+  L8.writeFile "test/example/bill_page_1.html" $ getResponseBody website
+
+-- I'm very sorry. Forgive me for my sins.
+fixNextPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' nextPage
+fixLastPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' lastPage
```

Comment:
The `Maybe` is actually really important -- it tells us whether or not we were able to find the link we want. If we haven't found the link, then we need to handle that somehow. The current implementation crashes the entire service, which is definitely not what you want.

Just work with the `Maybe` value until yuo know how to handle a `Nothing`. For example, this might be:

```haskell
someProcess websiteContents = do
  case getNextLink websiteContents of
    Nothing -> pure ()
    Just nextWebsiteUrl -> do
      newWebsiteContents <- download nextWebsiteUrl
      someProcess websiteContents
```

Reply:
This has been deleted